src/DatePickerInput.js

```diff
@@ -96,14 +96,22 @@ const DatePickerInput = React.createClass({
     }
   },
 
-  stopPropagation(e) {
-    if (this.props.closeOnClickOutside) {
-      e.stopPropagation();
+  getDatePickerInput() {
+    return React.findDOMNode(this.refs.datePickerInput);
+  },
+
+  isEventInsideDatePickerInput(el) {
+    if (el === this.getDatePickerInput()) {
+      return true;
+    } else if (el.parentNode) {
+      return this.isEventInsideDatePickerInput(el.parentNode);
+    } else {
+      return false;
     }
   },
 
-  hide() {
-    if (this.state.showing) {
+  hide(e) {
```

Comment:
This function is currently being used in other situations without an event here:
- https://github.com/buildo/rc-datepicker/blob/master/src/DatePickerInput.js#L124
- https://github.com/buildo/rc-datepicker/blob/master/src/DatePickerInput.js#L145

In these situations `e.target` will throw an error:
![image](https://cloud.githubusercontent.com/assets/4029499/13550326/74da9792-e31b-11e5-8bb1-57f61b02d566.png)

Maybe we should create a new function `hideOnClickOutside` (as we have for `hideOnEnterKey`) where we perform the new check you added (`isEventInsideDatePickerInput`) and then call the now safe `this.hide`. What do you think?